### PR TITLE
Add zobrist hashing and draw by threefold repetition.

### PR DESCRIPTION
**Pull request by edre**
Simple mechanism to prevent infinite games.

### CODE REVIEW

File: hive/src/hex.rs [hunk 1]
```diff
@@ -2,9 +2,9 @@ use std::collections::HashSet;
 
 #[derive(Copy, Clone, Debug, PartialEq, Eq, Hash, PartialOrd, Ord)]
 pub struct Hex {
-    pub x: i64,
-    pub y: i64,
-    pub z: i64,
+    pub x: i8,
```
Comment (wgreenberg) [inline hive/src/hex.rs:5]: i haven't done the math on it, but do you think `i8`s limit the possible boardspace significantly? i suspect you'd have to be playing pathologically to get outside `i8` bounds, but still

File: hive/src/sgf_parser.rs [hunk 2]
```diff
@@ -99,6 +99,6 @@ mod tests {
         std::fs::read_dir("./test_data")
             .expect("failed to open dir")
             .flat_map(|entry| entry)
-            .for_each(|entry| { read_sgf_file(entry.path()); });
+            .for_each(|entry| { read_sgf_file(dbg!(entry.path())); });
```
Comment (wgreenberg) [inline hive/src/sgf_parser.rs]: nitpick: unnecessary `dbg` here

**Comment by wgreenberg** [comment]
this looks great! i had one question about the switch to `i8`s, but otherwise down to merge this

File: hive/src/hex.rs (hunk 1 above)
Comment (edre) [reply hive/src/hex.rs:5]: There are only 28 tiles in the game, and in a normal game the entire hive doesn't migrate. It is possible to have two queens and two beetles slowly migrate an indefinite distance by crawling past each other, but this doesn't seem like an important usecase.

File: hive/src/sgf_parser.rs (hunk 2 above)
Comment (edre) [reply hive/src/sgf_parser.rs]: Removed.

**Comment by wgreenberg** [comment]
thanks @edre!

**Pull request merged**